Sources/NIOHTTP1/HTTPServerUpgradeHandler.swift

```diff
@@ -169,7 +169,10 @@ public final class HTTPServerUpgradeHandler: ChannelInboundHandler, RemovableCha
         // We'll attempt to upgrade. This may take a while, so while we're waiting more data can come in.
         self.upgradeState = .awaitingUpgrader
 
-        self.handleUpgrade(context: context, request: request, requestedProtocols: requestedProtocols).whenSuccess { callback in
+        self.handleUpgrade(context: context, request: request, requestedProtocols: requestedProtocols)
+            .hop(to: context.eventLoop) // the user might return a future from another EventLoop.
+            .whenSuccess { callback in
+            assert(context.eventLoop.inEventLoop)
```

Comment:
This indentation isn't great, can we clear that up?

Reply:
done, thanks